### PR TITLE
Make workspaces start using the "installer"

### LINKED ISSUE #6519: Make workspaces start

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR contains the necessary changes (and then some) to make workspaces start with an in-cluster configuration, i.e. using minio and the built-in registry.

## Related Issue(s)

Fixes #6519

## How to test
Run the installer against a fresh cluster, log in and start a workspace.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** and additionally assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

Associated issue: *#6519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/content-service-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service-api/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6550#issuecomment-959835742" title="Approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

Associated issue: *#6519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/content-service-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/image-builder-mk3/deployment.go
Comment (MrSimonEmms) [inline installer/pkg/components/image-builder-mk3/deployment.go:31]: Strictly speaking, this isn't going to be the correct hash - the `secret` function generates the secret each time it's run. What I'd propose is removing the `secret` (and probably `configmap`) call from the `objects.go` file and export them from the `deployment.go` file.

It would also make it a bit more efficient as the functions don't need to be generated twice then

**Review by MrSimonEmms — COMMENTED** [review]
I like the approach to the checksums - we just need to tweak it a little

**Review by MrSimonEmms — APPROVED** [review]
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d26c10b0df90bf14cf7d1b61e9f3fc2434926bc9</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6550#issuecomment-959835742" title="Approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6550#pullrequestreview-797514668" title="LGTM">MrSimonEmms</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

Associated issue: *#6519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/content-service-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service-api/OWNERS)~~ [csweichel]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [MrSimonEmms,csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6550#issuecomment-959835742" title="Approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6550#issuecomment-960594668" title="LGTM">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6550#pullrequestreview-797514668" title="LGTM">MrSimonEmms</a>*

Associated issue: *#6519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/content-service-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service-api/OWNERS)~~ [csweichel]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [MrSimonEmms,csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
